### PR TITLE
feat: Errors from gax layer

### PR DESCRIPTION
**Pull request by danieljbruce**
This PR contains tests for ensuring that errors which pass through the gax layer are presented to the user properly using unit tests from a grpc service.

### CODE REVIEW

File: test/util/mock-server.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      callback();
+    });
+    netServer.once('listening', () => {
+      // close the server if listening doesn't fail
+      netServer.close();
+    });
+    // netServer.listen(port);
+    netServer.listen(`localhost:${port}`);
+  }
+  describe('Ensure server shuts down properly when destroyed', () => {
```
Comment (danieljbruce) [inline test/util/mock-server.ts:51]: If I add a before hook with the following in it:

`checkPort(inputPort, false, done);`

the `assert.notEqual(err.code, 'EADDRINUSE');` fails, but I am not sure why at this time.

File: test/util/mock-server.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {MockServer} from '../../src/util/mock-servers/mock-server';
+import * as assert from 'assert';
+
+const tcpPortUsed = require('tcp-port-used');
+
+describe('Bigtable/Mock-Server', () => {
+  const inputPort = '1234';
+  let server: MockServer;
+  function checkPort(port: string, inUse: boolean, callback: () => void) {
+    tcpPortUsed.check(parseInt(port), 'localhost').then((isInUse: boolean) => {
```
Comment (bcoe) [inline test/util/mock-server.ts:29]: my advice would be to always use an async/await interface when possible, I think you could replace this with:

```
await checkPort()
```

**Review by bcoe — CHANGES_REQUESTED** [review]
Left a nit, I would always opt for an async/await appraoch, rather than Promise.then these days.

File: test/util/mock-server.ts (hunk 2 above)
Comment (danieljbruce) [reply test/util/mock-server.ts]: Done

Comment (danieljbruce) [reply test/util/mock-server.ts]: It actually looks like this now:

```
  async function checkPort(port: string, inUse: boolean, callback: () => void) {
    const isInUse: boolean = await tcpPortUsed.check(
      parseInt(port),
      'localhost'
    );
    assert.strictEqual(isInUse, inUse);
    callback();
  }
```

**Review by bcoe — APPROVED** [review]

**Pull request merged**